hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/operations/JoinShutdownTest.java

```diff
@@ -0,0 +1,186 @@
+        );
+        OperationAccessor.setCallId(op, 1);
+
+        ByteBuffer buffer = toBuffer(op);
+        int position = buffer.position();
+        upcast(buffer).position(286);
+        buffer.putInt(42_000_000);
+        upcast(buffer).position(position);
+        upcast(buffer).flip();
+
+        if (joinedBefore) {
+            node.getClusterService().resetJoinState();
+        }
+        send(buffer);
+        assertTrue(hz.getLifecycleService().isRunning());
+    }
+
+    @Test
+    public void joinedMemberShouldNotShutdown_whenConfigMismatchOp() throws Exception {
+        assertTrueEventually(() -> getNode(hz).getClusterService().isJoined());
+        if (joinedBefore) {
+            getNode(hz).getClusterService().resetJoinState();
+        }
+        send(new ConfigMismatchOp("Random reason"));
+        assertTrue(hz.getLifecycleService().isRunning());
+    }
+
+    @Test
+    public void joinedMemberShouldNotShutdown_whenBeforeJoinCheckFailureOp() throws Exception {
+        assertTrueEventually(() -> getNode(hz).getClusterService().isJoined());
+        if (joinedBefore) {
+            getNode(hz).getClusterService().resetJoinState();
+        }
+        send(new BeforeJoinCheckFailureOp("Random reason"));
+        assertTrue(hz.getLifecycleService().isRunning());
+    }
+
+    @Test
+    public void joinedMemberShouldNotShutdown_whenAuthenticationFailureOp() throws Exception {
+        assertTrueEventually(() -> getNode(hz).getClusterService().isJoined());
+        if (joinedBefore) {
+            getNode(hz).getClusterService().resetJoinState();
+        }
+        send(new AuthenticationFailureOp());
+        assertTrue(hz.getLifecycleService().isRunning());
+    }
+
+    private static void send(Operation operation) throws Exception {
+        send((ByteBuffer) upcast(toBuffer(operation)).flip());
+    }
+
+    private static void send(ByteBuffer buffer) throws Exception {
+        // send bytebuffer
+        try (SocketChannel socketChannel = SocketChannel.open()) {
+            socketChannel.connect(new InetSocketAddress("127.0.0.1", 5701));
+            while (buffer.hasRemaining()) {
+                socketChannel.write(buffer);
+            }
+        }
+
+        // sleep for member to process the operation
+        Thread.sleep(100);
+    }
+
+    private static ByteBuffer toBuffer(Operation operation) {
+        // op -> packet
+        Data data = toData(operation);
+        Packet packet = new Packet(data.toByteArray());
+        packet.setPacketType(Packet.Type.OPERATION);
+
+        //packet -> bytebuffer
+        ByteBuffer buffer = ByteBuffer.allocate(packet.getFrameLength() + 3);
+        buffer.put("HZC".getBytes());
+        PacketIOHelper packetIOHelper = new PacketIOHelper();
+        packetIOHelper.writeTo(packet, buffer);
+        return buffer;
+    }
+
+    @Override
+    protected Config getConfig() {
+        return smallInstanceConfigWithoutJetAndMetrics();
+    }
```

Comment:
IMO, not needed

Reply:
https://github.com/hazelcast/hazelcast/pull/24577/commits/3be8d7c915b677151630184e9e3b8b89bc3cd041